mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/SocketPolicy.java

```diff
@@ -50,6 +50,9 @@
    */
   DISCONNECT_AFTER_REQUEST,
 
+  /** Close connection after reading half of the request body (if present). */
```

Comment:
Wouldn't hurt! This entire class is so painfully organic in design. I hope to someday make a do-over where we borrow models from the client.